File: lecture_1/cash/cash.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>

int main(void){
    float change;
    int coins = 0;

    do{
       change = get_float("Change owed: ");
    } while (change && change < 0);

    if(change > 0){
        int cents = round(change * 100);
        int coins25 = cents / 25;
        coins += coins25;

        int change25 = cents % 25;

        if(change25 > 0 ){
            int coins10 = change25 / 10;
            coins += coins10;
            int change10 = change25 % 10;

            if(change10 > 0 ){
                int coins5 = change10 / 5;
                coins += coins5;
                int change5 = change10 % 5;
                coins += change5;
            }
        }
    }

    printf("%i\n", coins);
}

File: lecture_6/cash/cash.py
from cs50 import get_float

coins = 0

while True:
    change = get_float("Change owed: ")

    if change and change > 0:
        break

cents = round(change * 100)
coins25 = cents // 25
coins += coins25

# remainder of the division by 25
change25 = cents % 25

if change25 > 0:
    coins10 = change25 // 10
    coins += coins10

    # remainder of the division by 10
    change10 = change25 % 10

    if change10 > 0:
        coins5 = change10 // 5
        coins += coins5
        change5 = change10 % 5

        coins += change5

print(coins)